### PR TITLE
Onboarding authorization for autogeneration

### PR DESCRIPTION
**Pull request by dagoroz**
```

> azure-schema-generator@1.0.0 generate-single D:\Repos\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "authorization/resource-manager"

[C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc] executing: git fetch -q
[C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
Using autogenlist config:
{
  "basePath": "authorization/resource-manager",
  "namespace": "Microsoft.Authorization",
  "resourceConfig": [
    {
      "type": "roleAssignments",
      "scopes": 30
    },
    {
      "type": "roleDefinitions",
      "scopes": 30
    }
  ]
}
[D:\Repos\azure-resource-manager-schemas\generator] executing: autorest.cmd --version=3.0.6320 --use=@autorest/azureresourceschema@3.0.92 --azureresourceschema --output-folder=C:\Users\DAOROZ~1.RED\AppData\Local\Temp\ymci4l1m9j --multiapi --title=none --pass-thru:subset-reducer C:\Users\DAOROZ~1.RED\AppData\Local\Temp\schm_azspc\specification\authorization\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6247; node: v12.17.0, max-memory: 2048 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\daorozco.REDMOND\.autorest\@autorest_core@3.0.6320\node_modules\@autorest\core\dist' (3.0.6320)
   Installing AutoRest extension '@autorest/azureresourceschema' (3.0.92)
   Installed AutoRest extension '@autorest/azureresourceschema' (3.0.92->3.0.92)
Processing batch task - {"tag":"schema-authorization-2020-04-01-preview"} .
WARNING: Skipping path '/{roleId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-authorization-2018-09-01-preview"} .
WARNING: Skipping path '/{roleId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-authorization-2018-07-01-preview"} .
Processing batch task - {"tag":"schema-authorization-2018-01-01-preview"} .
WARNING: Skipping path '/{roleId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-authorization-2017-10-01-preview"} .
WARNING: Skipping path '/{roleAssignmentId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-authorization-2015-07-01"} .
WARNING: Skipping path '/{roleAssignmentId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-authorization-2015-06-01"} .
[15.7 s] Generation Complete
================================================================================================================================
Filename: D:\Repos\azure-resource-manager-schemas\schemas\2015-07-01\Microsoft.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2015-07-01
Resource Types (Tenant Scope):
- roleAssignments
- roleDefinitions
Resource Types (Management Group Scope):
- roleAssignments
- roleDefinitions
Resource Types (Subscription Scope):
- roleAssignments
- roleDefinitions
Resource Types (Resource Group Scope):
- roleAssignments
- roleDefinitions
================================================================================================================================
================================================================================================================================
Filename: D:\Repos\azure-resource-manager-schemas\schemas\2017-10-01-preview\Microsoft.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2017-10-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
================================================================================================================================
================================================================================================================================
Filename: D:\Repos\azure-resource-manager-schemas\schemas\2018-01-01-preview\Microsoft.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2018-01-01-preview
Resource Types (Tenant Scope):
- roleAssignments
- roleDefinitions
Resource Types (Management Group Scope):
- roleAssignments
- roleDefinitions
Resource Types (Subscription Scope):
- roleAssignments
- roleDefinitions
Resource Types (Resource Group Scope):
- roleAssignments
- roleDefinitions
================================================================================================================================
================================================================================================================================
Filename: D:\Repos\azure-resource-manager-schemas\schemas\2018-09-01-preview\Microsoft.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2018-09-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
================================================================================================================================
================================================================================================================================
Filename: D:\Repos\azure-resource-manager-schemas\schemas\2020-04-01-preview\Microsoft.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2020-04-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
================================================================================================================================

```

### CODE REVIEW

**Comment by dagoroz** [comment]
@anthony-c-martin 
Sorry for the random ping. I tried to follow the repo instructions to the best of my ability but I don't knwo what (if anything) still needs to be done.

**Comment by dagoroz** [comment]
@apclouds Sorry for the random ping. I tried to follow the repo instructions to the best of my ability but I don't knwo what (if anything) still needs to be done.

**Comment by apclouds** [comment]
@dagoroz will take a look on our end.

**Comment by dagoroz** [comment]
@apclouds follow up ping

**Comment by leni-msft** [comment]
@dagoroz Could you rebase master branch, run the `generate-single` again, and then update the PR description? since we recently changed parameters of generating schemas.

**Review by leni-msft — COMMENTED** [review]
@dagoroz Daniel Orozco FTE Could you rebase master branch, run the generate-single again, and then update the PR description? since we recently changed parameters of generating schemas.

**Comment by dagoroz** [comment]
@leni-msft I rebased master and updated PR

**Comment by leni-msft** [comment]
@dagoroz I once tried to onboard authorization for autogeneration, and seems some resource types were unable to determine the scope. Hence needing `resourceConfig` (see https://github.com/Azure/azure-resource-manager-schemas/pull/893/commits/db6f22105bef17682bb6fbe4d575f73866e1b92e#diff-e73a42dac3b7687d870c8e14015e427bR23-R36 ) Please confirm if the resource scope is correct, and regenerate the schemas. As for test failures, please refer to https://github.com/Azure/azure-resource-manager-schemas/blob/master/README.md#runing-unit-tests to run local. then you should be able to find the error logs.

**Comment by dagoroz** [comment]
@leni-msft I confirmed scope is correct and regenerated the schemas
I ran the tests locally and this is the error I get. (attached at the end)

This seems to be an issue with Microsoft.Networking that we simply don't touch. What should be done in these cases?

```
1) Validate test templates against VSCode language service
       running schema validation on 'D:\Repos\azure-resource-manager-schemas\tools\templateTests\Template_2.2019-04-01.json':

      AssertionError: expected [ Array(3) ] to deeply equal []
      + expected - actual

      -[
      -  {
      -    "message": "Missing property \"location\"."
      -    "range": {
      -      "end": {
      -        "character": 5
      -        "line": 1274
      -      }
      -      "start": {
      -        "character": 4
      -        "line": 1274
      -      }
      -    }
      -    "severity": 2
      -  }
      -  {
      -    "code": 1
      -    "message": "Value is not accepted. Valid values: \"Microsoft.Network/publicIPAddresses\"."
      -    "range": {
      -      "end": {
      -        "character": 45
      -        "line": 1277
      -      }
      -      "start": {
      -        "character": 14
      -        "line": 1277
      -      }
      -    }
      -    "severity": 2
      -  }
      -  {
      -    "message": "Missing property \"publicIPAllocationMethod\"."
      -    "range": {
      -      "end": {
      -        "character": 18
      -        "line": 1278
      -      }
      -      "start": {
      -        "character": 6
      -        "line": 1278
      -      }
      -    }
      -    "severity": 2
      -  }
      -]
      +[]

      at Context.<anonymous> (tests.ts:224:30)
```

File: schemas/2019-04-01/deploymentTemplate.json [hunk 1]
```diff
@@ -221,7 +221,6 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2018-07-01/Microsoft.Media.json#/resourceDefinitions/mediaServices_streamingPolicies" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-07-01/Microsoft.Media.json#/resourceDefinitions/mediaServices_transforms" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-07-01/Microsoft.Media.json#/resourceDefinitions/mediaServices_transforms_jobs" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2016-09-01/Microsoft.Authorization.json#/resourceDefinitions/locks" },
```
Comment (leni-msft) [inline schemas/2019-04-01/deploymentTemplate.json:224]: The error should be caused by deleting this line. With the `2016-09-01`, schema validator find one possible valid value is `Microsoft.Network/publicIPAddresses`, and then use it to validate.

Comment (dagoroz) [reply schemas/2019-04-01/deploymentTemplate.json]: I reverted that change and it seemed to fix the issue.
All tests are passing again

Comment (dagoroz) [reply schemas/2019-04-01/deploymentTemplate.json]: @leni-msft forgot to @

**Review by leni-msft — APPROVED** [review]

File: schemas/2019-04-01/deploymentTemplate.json (hunk 1 above)
Comment (leni-msft) [reply schemas/2019-04-01/deploymentTemplate.json]: ok, LGTM

**Comment by dagoroz** [comment]
@leni-msft
Do you need any further action from my part for merging?
or can I consider this as complete?

**Comment by leni-msft** [comment]
@anthony-c-martin @jorgecotillo pls help check & merge this PR.

**Comment by dagoroz** [comment]
@anthony-c-martin @jorgecotillo @leni-msft 
ping

**Review by anthony-c-martin — APPROVED** [review]

**Comment by anthony-c-martin** [comment]
@dagoroz Sorry for the delay on getting this merged.

**Pull request merged**